Cswatt/operator2 (#8494)
DataDog/documentation: content/en/agent/kubernetes/operator_configuration.md

| agent.additionalLabels                                                                                     | AdditionalLabels provide labels that will be added to the cluster checks runner Pods.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

> | agent.apm.enabled                                                                                          | Enable this to enable APM and tracing, on port 8126 ref: https://github.com/DataDog/docker-dd-agent#tracing-from-the-host                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer: same comment about making these actual clickable URLs. You also use periods inconsistently in this table too.

| agent.apm.env                                                                                              | The Datadog Agent supports many environment variables Ref: https://docs.datadoghq.com/agent/docker/?tab=standard#environment-variables                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | agent.apm.hostPort                                                                                         | Number of port to expose on the host. If specified, this must be a valid port number, 0 < x < 65536. If HostNetwork is specified, this must match ContainerPort. Most containers do not need this.                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer: ```suggestion
| agent.apm.hostPort                                                                                         | Number of ports to expose on the host. If specified, this must be a valid port number, 0 < x < 65536. If HostNetwork is specified, this must match ContainerPort. Most containers do not need this.                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```

| agent.config.checksd.configMapName                                                                         | ConfigMapName name of a ConfigMap used to mount a directory                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |

> | agent.config.collectEvents                                                                                 | nables this to start event collection from the kubernetes API ref: https://docs.datadoghq.com/agent/kubernetes/event_collection/                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| agent.config.collectEvents                                                                                 | Enable this to start event collection from the kubernetes API ref: https://docs.datadoghq.com/agent/kubernetes/event_collection/                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```

| agent.config.confd.configMapName                                                                           | ConfigMapName name of a ConfigMap used to mount a directory                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |

> | agent.config.criSocket.criSocketPath                                                                       | Path to the container runtime socket (if different from Docker) This is supported starting from agent 6.6.0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |

Reviewer: ```suggestion
| agent.config.criSocket.criSocketPath                                                                       | Path to the container runtime socket (if different from Docker). This is supported starting from agent 6.6.0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
```

> | agent.config.criSocket.dockerSocketPath                                                                    | Path to the docker runtime socket                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| agent.config.criSocket.dockerSocketPath                                                                    | Path to the Docker runtime socket                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
```

| agent.config.dogstatsd.dogstatsdOriginDetection                                                            | Enable origin detection for container tagging https://docs.datadoghq.com/developers/dogstatsd/unix_socket/#using-origin-detection-for-container-tagging                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | agent.config.dogstatsd.useDogStatsDSocketVolume                                                            | Enable dogstatsd over Unix Domain Socket ref: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |

Reviewer: ```suggestion
| agent.config.dogstatsd.useDogStatsDSocketVolume                                                            | Enable DogStatsD over Unix Domain Socket ref: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```

| agent.config.env                                                                                           | The Datadog Agent supports many environment variables Ref: https://docs.datadoghq.com/agent/docker/?tab=standard#environment-variables                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | agent.config.hostPort                                                                                      | Number of port to expose on the host. If specified, this must be a valid port number, 0 < x < 65536. If HostNetwork is specified, this must match ContainerPort. Most containers do not need this.                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer: ```suggestion
| agent.config.hostPort                                                                                      | Number of ports to expose on the host. If specified, this must be a valid port number, 0 < x < 65536. If HostNetwork is specified, this must match ContainerPort. Most containers do not need this.                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```

| agent.config.leaderElection                                                                                | Enables leader election mechanism for event collection.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | agent.config.logLevel                                                                                      | Set logging verbosity, valid log levels are: trace, debug, info, warn, error, critical, and off                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| agent.config.logLevel                                                                                      | Set logging verbosity, valid log levels are: `trace`, `debug`, `info`, `warn`, `error`, `critical`, and `off`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```

> | agent.config.podAnnotationsAsTags                                                                          | Provide a mapping of Kubernetes Annotations to Datadog Tags. <KUBERNETES_ANNOTATIONS>: <DATADOG_TAG_KEY>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| agent.config.podAnnotationsAsTags                                                                          | Provide a mapping of Kubernetes Annotations to Datadog Tags. `<KUBERNETES_ANNOTATIONS>: <DATADOG_TAG_KEY>`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```

> | agent.config.podLabelsAsTags                                                                               | Provide a mapping of Kubernetes Labels to Datadog Tags. <KUBERNETES_LABEL>: <DATADOG_TAG_KEY>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| agent.config.podLabelsAsTags                                                                               | Provide a mapping of Kubernetes Labels to Datadog Tags. `<KUBERNETES_LABEL>: <DATADOG_TAG_KEY>`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```

| agent.config.resources.requests                                                                            | Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/                                                                                                                                                                                                                                                                                                                                     |

> | agent.config.securityContext.allowPrivilegeEscalation                                                      | AllowPrivilegeEscalation controls whether a process can gain more privileges than its parent process. This bool directly controls if the no_new_privs flag will be set on the container process. AllowPrivilegeEscalation is true always when the container is: 1) run as Privileged 2) has CAP_SYS_ADMIN                                                                                                                                                                                                                                                                                                                                              |

Reviewer: bool isn't right but I am not sure what it should be. tool?

| agent.config.securityContext.privileged                                                                    | Run container in privileged mode. Processes in privileged containers are essentially equivalent to root on the host. Defaults to false.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | agent.config.securityContext.procMount                                                                     | procMount denotes the type of proc mount to use for the containers. The default is DefaultProcMount which uses the container runtime defaults for readonly paths and masked paths. This requires the ProcMountType feature flag to be enabled.                                                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer: I just realized that many of these are not in code ticks, so `procMount`, `DefaultProcMount`, etc, through this whole table

| agent.customConfig.configMap.fileKey                                                                       | FileKey corresponds to the key used in the ConfigMap.Data to store the configuration file content                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |

> | agent.customConfig.configMap.name                                                                          | Name the ConfigMap name                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| agent.customConfig.configMap.name                                                                          | Name of the ConfigMap name                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```

| agent.deploymentStrategy.reconcileFrequency                                                                | The reconcile frequency of the ExtendDaemonSet                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

> | agent.deploymentStrategy.rollingUpdate.maxParallelPodCreation                                              | The maxium number of pods created in parallel. Default value is 250.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| agent.deploymentStrategy.rollingUpdate.maxParallelPodCreation                                              | The maximum number of pods created in parallel. Default value is 250.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
```

> | agent.deploymentStrategy.rollingUpdate.maxPodSchedulerFailure                                              | MaxPodSchedulerFailure the maxinum number of not scheduled on its Node due to a scheduler failure: resource constraints. Value can be an absolute number (ex: 5) or a percentage of total number of DaemonSet pods at the start of the update (ex: 10%). Absolute                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: I think this needs rephrased - what does it mean to be the maximum number of not scheduled?

| agent.log.openFilesLimit                                                                                   | Set the maximum number of logs files that the Datadog Agent will tail up to. Increasing this limit can increase resource consumption of the Agent. ref: https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#log-collection-setup Default to 100                                                                                                                                                                                                                                                                                                                                                                                             |

> | agent.log.podLogsPath                                                                                      | This to allow log collection from pod log path. Default to /var/log/pods                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |

Reviewer: ```suggestion
| agent.log.podLogsPath                                                                                      | Set this to allow log collection from pod log path. Default to `/var/log/pods`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```

> | agent.log.tempStoragePath                                                                                  | This path (always mounted from the host) is used by Datadog Agent to store information about processed log files. If the Datadog Agent is restarted, it allows to start tailing the log files from the right offset Default to /var/lib/datadog-agent/logs                                                                                                                                                                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| agent.log.tempStoragePath                                                                                  | This path (always mounted from the host) is used by the Datadog Agent to store information about processed log files. If the Datadog Agent is restarted, it allows to start tailing the log files from the right offset Default to `/var/lib/datadog-agent/logs`                                                                                                                                                                                                                                                                                                                                                                                          |
```

| agent.rbac.serviceAccountName                                                                              | Used to set up the service account name to use Ignored if the field Create is true                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

> | agent.systemProbe.appArmorProfileName                                                                      | AppArmorProfileName specify a apparmor profile                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| agent.systemProbe.appArmorProfileName                                                                      | Use `AppArmorProfileName` to specify a `apparmor` profile.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```

> | agent.systemProbe.bpfDebugEnabled                                                                          | BPFDebugEnabled logging for kernel debug                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| agent.systemProbe.bpfDebugEnabled                                                                          | Use BPFDebugEnabled logging to debug your kernel.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```

| agent.config.securityContext.windowsOptions.runAsUserName                                                  | The `UserName` in Windows to run the entrypoint of the container process. Defaults to the user specified in image metadata if unspecified. May also be set in `PodSecurityContext`. If set in both `SecurityContext` and `PodSecurityContext`, the value specified in `SecurityContext` takes precedence. This field is beta-level and may be disabled with the `WindowsRunAsUserName` feature flag.                                                                                                                                                                                                                                                               |

> | agent.config.tags                                                                                          | List of tags to attach to every metric, event, and service check collected by this Agent. See the [Tagging documentation][8]                                                                                                                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| agent.config.tags                                                                                          | List of tags to attach to every metric, event, and service check collected by this Agent. See the [Tagging documentation][8].                                                                                                                                                                                                                                                                                                                                                                                                        |
```

| agent.deploymentStrategy.rollingUpdate.maxParallelPodCreation                                              | The maxium number of pods created in parallel. Default value is 250.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |

> | agent.deploymentStrategy.rollingUpdate.maxPodSchedulerFailure                                              | 	`maxPodSchedulerFailure` is the maxinum number of pods scheduled on its Node due to a scheduler failure: resource constraints. Value can be an absolute number (ex: 5) or a percentage of total number of DaemonSet pods at the start of the update (ex: 10%). Absolute.                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| agent.deploymentStrategy.rollingUpdate.maxPodSchedulerFailure                                              | 	`maxPodSchedulerFailure` is the maximum number of pods scheduled on its Node due to a scheduler failure: resource constraints. Value can be an absolute number (ex: 5) or a percentage of total number of DaemonSet pods at the start of the update (ex: 10%). Absolute.                                                                                                                                                                                                                                                                                                                                                                                      |
```

| agent.dnsConfig.nameservers                                                                                | A list of DNS name server IP addresses. This are appended to the base nameservers generated from `dnsPolicy`. Duplicated nameservers are removed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | agent.dnsConfig.options                                                                                    | A list of DNS resolver options. This are merged with the base options generated from `dnsPolicy`. Duplicated entries will be removed. Resolution options given in `options` override those that appear in the base `dnsPolicy`.                                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer: ```suggestion
| agent.dnsConfig.options                                                                                    | A list of DNS resolver options. These are merged with the base options generated from `dnsPolicy`. Duplicated entries will be removed. Resolution options given in `options` override those that appear in the base `dnsPolicy`.                                                                                                                                                                                                                                                                                                                                                                                                                     |
```

| agent.hostPID                                                                                              | Use the host's PID namespace. Optional: Defaults to `false`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

> | agent.image.name                                                                                           | Define the image to use Use `datadog/agent:latest` for Datadog Agent 6. Use `datadog/dogstatsd:latest` for stand-alone Datadog Agent DogStatsD. Use `datadog/cluster-agent:latest` for Datadog Cluster Agent.                                                                                                                                                                                                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| agent.image.name                                                                                           | Define the image to use `datadog/agent:latest` for Datadog Agent 6. Use `datadog/dogstatsd:latest` for stand-alone Datadog Agent DogStatsD. Use `datadog/cluster-agent:latest` for Datadog Cluster Agent.                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```

| agent.image.pullPolicy                                                                                     | The Kubernetes pull policy. Use `Always`, `Never`, or `IfNotPresent`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

> | agent.image.pullSecrets                                                                                    | It is possible to specify Docker registry credentials. [See the Kubernetes documentation][9].                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| agent.image.pullSecrets                                                                                    | Specifies the Docker registry credentials. [See the Kubernetes documentation][9].                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```

| agent.log.containerCollectUsingFiles                                                                       | Collect logs from files in `/var/log/pods` instead of using container runtime API. This is usually the most efficient way of collecting logs. See the [Log Collection][10] documentation. Default: `true`.                                                                                                                                                                                                                                                                                                                                                                                                       |

> | agent.log.containerLogsPath                                                                                | Allow log collection from container log path. Set to a different path if not using docker runtime. See the [Kubernetes documentation][11].  Defaults to `/var/lib/docker/containers`.                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| agent.log.containerLogsPath                                                                                | Allow log collection from the container log path. Set to a different path if not using docker runtime. See the [Kubernetes documentation][11].  Defaults to `/var/lib/docker/containers`.                                                                                                                                                                                                                                                                                                                                                                                                       |
```

| agent.rbac.serviceAccountName                                                                              | Used to set up the service account name to use `Ignored` if the field `Create` is true.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

> | agent.systemProbe.appArmorProfileName                                                                      | Specify a AppArmor profile.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| agent.systemProbe.appArmorProfileName                                                                      | Specify an AppArmor profile.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```

| agent.systemProbe.securityContext.windowsOptions.gmsaCredentialSpecName                                    | `GMSACredentialSpecName` is the name of the GMSA credential spec to use. This field is alpha-level and is only honored by servers that enable the WindowsGMSA feature flag.                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

> | agent.systemProbe.securityContext.windowsOptions.runAsUserName                                             | The `UserName` in Windows to run the entrypoint of the container process. Defaults to the user specified in image metadata if unspecified. May also be set in `PodSecurityContext`. If set in both `SecurityContext` and `PodSecurityContext`, the value specified in `SecurityContext` takes precedence. This field is beta-level and may be disabled with the `WindowsRunAsUserName` feature flag.                                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| agent.systemProbe.securityContext.windowsOptions.runAsUserName                                             | Use the `UserName` in Windows to run the entry point of the container process. Defaults to the user specified in image metadata if unspecified. May also be set in `PodSecurityContext`. If set in both `SecurityContext` and `PodSecurityContext`, the value specified in `SecurityContext` takes precedence. This field is beta-level and may be disabled with the `WindowsRunAsUserName` feature flag.                                                                                                                                                                                                                                                               |
```

| clusterAgent.image.pullPolicy                                                                              | The Kubernetes pull policy. Use `Always`, `Never`, or `IfNotPresent`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |

> | clusterAgent.image.pullSecrets                                                                             | It is possible to specify Docker registry credentials. See the [Kubernetes documentation][9].                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| clusterAgent.image.pullSecrets                                                                             | Specifies Docker registry credentials. See the [Kubernetes documentation][9].                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
```
